### PR TITLE
[6/n Update test rpc path

### PR DESCRIPTION
**Pull request by tktrungna**
Fixes #62380

* update `test_rpc` function to call wheel install folder {sitepackages}/torch instead of build/ folder
* add IN_WHEEL_TEST to limit the change for linux CI GHA only

### Test plan
check if all ci workflows pass

### CODE REVIEW

File: .jenkins/pytorch/test.sh [hunk 1]
```diff
@@ -307,12 +308,29 @@ test_distributed() {
...
     # NB: the ending test_rpc must match the current function name for the current
     # test reporting process (in print_test_stats.py) to function as expected.
     TEST_REPORTS_DIR=test/test-reports/cpp-rpc/test_rpc
     mkdir -p $TEST_REPORTS_DIR
-    build/bin/test_cpp_rpc --gtest_output=xml:$TEST_REPORTS_DIR/test_cpp_rpc.xml
+    "$TEST_BASE_DIR"/test_cpp_rpc --gtest_output=xml:$TEST_REPORTS_DIR/test_cpp_rpc.xml
+
+    if [[ -n "$IN_WHEEL_TEST" ]]; then
+      # Restore the build folder to avoid any impact on other tests
+      ls -R build_renamed
```
Comment (walterddr) [inline .jenkins/pytorch/test.sh:330]: what was this for?

**Review by walterddr — APPROVED** [review]
looks good. quick note on a possible noise in print out

File: .jenkins/pytorch/test.sh (hunk 1 above)
Comment (tktrungna) [reply .jenkins/pytorch/test.sh:330]: oops, some debug code I forgot to clean. Updated

File: .jenkins/pytorch/test.sh [hunk 2]
```diff
@@ -307,12 +308,27 @@ test_distributed() {
 
 test_rpc() {
   if [[ "$BUILD_ENVIRONMENT" != *rocm* ]]; then
-    echo "Testing RPC C++ tests"
+    if [[ -n "$IN_WHEEL_TEST" ]]; then
+      echo "Testing RPC C++ tests with the install folder"
+      # Rename the build folder when running test to ensure it
+      # is not depended on the folder
+      mv "$BUILD_DIR" "$BUILD_RENAMED_DIR"
```
Comment (janeyx99) [inline .jenkins/pytorch/test.sh:315]: Where is $BUILD_RENAMED_DIR defined? I'm failing to find it in your changes--does it come from exisitng code?

Comment (walterddr) [reply .jenkins/pytorch/test.sh:315]: yes it was from the [1/n] PR

Comment (seemethere) [reply .jenkins/pytorch/test.sh:320]: Would these be better served as a `ghstack` PR so we can get a full lineage of these changes?

Just reference for the future

**Review by seemethere — APPROVED** [review]

File: .jenkins/pytorch/test.sh (hunk 2 above)
Comment (tktrungna) [reply .jenkins/pytorch/test.sh:354]: > Would these be better served as a `ghstack` PR so we can get a full lineage of these changes?
> 
> Just reference for the future

Thanks, will do in next diffs

**Pull request closed without merging**